portality/scripts/anon_import.py

```diff
@@ -0,0 +1,94 @@
+import esprit, codecs, json
+from portality.core import app, initialise_index
+from portality.store import StoreFactory
+from portality import constants
+
+
+def do_import(config):
+    host = app.config["ELASTIC_SEARCH_HOST"]
+    index = app.config["ELASTIC_SEARCH_DB"]
+    if config.get("elastic_search_host") is not None:
+        host = config.get("elastic_search_host")
+        app.config["ELASTIC_SEARCH_HOST"] = host
+    if config.get("elastic_search_db") is not None:
+        index = config.get("elastic_search_db")
+        app.config["ELASTIC_SEARCH_DB"] = index
+
+    print("\n")
+    print("Using host {x} and index {y}\n".format(x=host, y=index))
+    conn = esprit.raw.make_connection(None, host, None, index)
+
+    # filter for the types we are going to work with
+    import_types = {}
+    for t, s in config.get("types", {}).iteritems():
+        if s.get("import", False) is True:
+            import_types[t] = s
+
+    print("==Carrying out the following import==")
+    for import_type, cfg in import_types.iteritems():
+        count = "All" if cfg.get("limit", -1) == -1 else cfg.get("limit")
+        print("{x} from {y}".format(x=count, y=import_type))
+    print("\n")
+
+    if config.get("confirm", True):
+        text = raw_input("Continue? [y/n]")
+        if text.lower() != "y":
+            exit()
+
+    # remove all the types that we are going to import
+    for import_type in import_types.keys():
+        esprit.raw.delete(conn, import_type)
+
+    # re-initialise the index (sorting out mappings, etc)
+    print("==Initialising Index for Mappings==")
+    initialise_index(app)
+
+    mainStore = StoreFactory.get("anon_data")
+    tempStore = StoreFactory.tmp()
+    container = constants.STORE_ANON_DATA
+
+    print("\n==Importing==")
+    for import_type, cfg in import_types.iteritems():
+        count = "all" if cfg.get("limit", -1) == -1 else cfg.get("limit")
+        print("Importing {x} from {y}".format(x=count, y=import_type))
+        print("Obtaining {x} from storage".format(x=import_type))
+
+        limit = cfg.get("limit", -1)
+        limit = None if limit == -1 else limit
+
+        n = 1
+        while True:
+            filename = import_type + ".bulk" + "." + str(n)
+            handle = mainStore.get(container, filename)
+            if handle is None:
+                break
+            print("Retrieved {x} from storage".format(x=filename))
+            tempStore.store(container, filename, source_stream=handle)
```

Comment:
We don't have any specific exception handling here, are we happy for the whole import to fail if storage has a blip?

Reply:
For the time being, yes.  The storage implementation is weakly tested at this stage and we decided that we'd test it properly when we want to use it for something more serious and it gets a proper implementation.  In the mean time this developer-oriented script is allowed to be a bit more fragile, I think.